### DISCUSSION TITLE
gogs bat/rest?

### DISCUSSION THREAD

**Discussion by feiyve** [gogs/gogs#6826] [Q&A]
Hi@all

i come from "java eclipse maven svn" environment and startet 1 week ago with git and gogs, fast learning including some heart-attacks (reset --hard: where are my fiiiiles???) =)
now i want to setup gogs with some repos, many files, some teams and some users per team (we want to migrate completely fom svn to git), all with different permissions to see/pull/push on specific repos/branches or not and so on, And perhaps in 2 days another gogs-server with nearly the same configuration, but perhaps with some submodules and slightly other configuration/permissions and so on. Would be a little bit uncool to click and type everything together in gogs-ui in browser.

Is ist possible to fully administer a gogs-installation per windows bat oder REST-Calls in Browser or something like that?

Greetings from germany

**Comment by feiyve** [comment]
colleague found rest-api on google, i'll give it a try =)

**Reply by unknwon** [reply]
The current state of Gogs's API is not ~~super~~ powerful to be frank 😂 But... let's see!

**Comment by feiyve** [comment]
Brain is growing constantly. But how can one delete an organization with curl? Is it available in the api?

We want to separate our master-development from the specific customer-releases (not branches, all get the same code and we control it with data in database oder parameters and so on, but not all customers have the same release at the same time, so wie make snapshots to be able to hotfix everything everytime). Therefore i tried to use an separate organization for each customer and in every organization then the different repos.
Create organization, create repo, add members to repo all works fine, even delete repo from organization works (did'nt find it in api-docs), but how can i curl-gogs delete whole organization?

**Reply by unknwon** [reply]
> Brain is growing constantly. But how can one delete an organization with curl? Is it available in the api?

It is intentionally not exposed by API, and requires entering the user password through UI. But if you really want to achieve this via curl, you can modify the code in https://github.com/gogs/gogs/blob/main/internal/route/api/v1/org/org.go and call https://github.com/gogs/gogs/blob/1791665f744b4fccf263b182e5f0fa2f2a195610/internal/route/org/setting.go#L121

**Reply by feiyve** [reply]
uuuääääh ....hmmmm .... errrr ... xD
ok, i give it a try @home. It's then my first try in contributing on GitHub and i never saw "go" before =)

**Reply by unknwon** [reply]
😂 Or if you're willing to wait (for god knows how long), you can make a [feature request](https://github.com/gogs/gogs/issues/new?assignees=&labels=%F0%9F%8E%AF+feature&template=feature_request.yml).
